packages/server/src/api/controllers/role.ts

```diff
@@ -143,4 +144,20 @@ export async function accessible(ctx: UserCtx) {
   } else {
     ctx.body = await roles.getUserRoleIdHierarchy(roleId!)
   }
+
+  // If a custom role is provided in the header, filter out higher level roles
+  const roleHeader = ctx.header?.["x-budibase-role"] as string
```

Comment:
I think we have an enumeration of headers somewhere - I think in backend-core `src/constants/misc.ts` - should probably move this to types for easy - but if you could use the `PREVIEW_ROLE` from that would be great!